src/main/java/io/dapr/components/StateStore.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2022 The Dapr Authors
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.dapr.components;
+
+import io.dapr.components.aspects.AdvertisesFeatures;
+import io.dapr.components.aspects.InitializableWithProperties;
+import lombok.Builder;
+import lombok.NonNull;
+
+import java.util.Optional;
+
+public interface StateStore extends InitializableWithProperties, AdvertisesFeatures {
+  Optional<StateStore.Value> get(@NonNull String key);
+
+  void delete(@NonNull String key, @NonNull String etag);
+
+  void set(@NonNull String key, @NonNull StateStore.Value value);
+
+  @lombok.Value
+  @Builder
+  public static class Value {
+    @NonNull byte[] data;
+    @NonNull String etag;
+  }
```

Comment:
nit (not necessary as it is sandboxed code):

Add javadocs

Reply:
Didn't get the sandboxed comment